go.mod: Update gvisor-tap-vsock to latest release

cfergeau opened the pull request:
This brings go 1.20/1.21 support, but drops go 1.19 support.
After https://github.com/crc-org/crc/pull/3843 this should be fine.

This fixes this build error with go 1.21:
```
package github.com/crc-org/crc/v2/cmd/crc
        imports github.com/crc-org/crc/v2/cmd/crc/cmd
        imports github.com/containers/gvisor-tap-vsock/pkg/virtualnetwork
        imports github.com/containers/gvisor-tap-vsock/pkg/services/dhcp
        imports github.com/containers/gvisor-tap-vsock/pkg/tap
        imports gvisor.dev/gvisor/pkg/tcpip/stack
        imports gvisor.dev/gvisor/pkg/sync/locking
        imports gvisor.dev/gvisor/pkg/gohacks: build constraints exclude all Go files in /Users/yevhen/work/redhat/crc/vendor/gvisor.dev/gvisor/pkg/gohacks
make: *** [install] Error 1
```



**Fixes:** Issue #N

**Relates to:** Issue #N, PR #N, ...

## Solution/Idea

Describe in plain English what you solved and how. For instance, _Added `start` command to CRC so the user can create a VM and set-up a single-node OpenShift cluster on it with one command. It requires blablabla..._

## Proposed changes

List main as well as consequential changes you introduced or had to introduce.

1. Add `start` command.
2. Add `setup` as prerequisite to `start`.
3. ...

## Testing

What is the _bottom-line_ functionality that needs testing? Describe in pseudo-code or in English. Use verifiable statements that tie your changes to existing functionality.

1. `start` succeeds first time after `setup` succeeded
2. stdout contains ... if `start` succeeded
3. stderr contains ... after `start` failed
4. `status` returns ... if `start` succeeded
5. `status` returns ... if `start` failed
6. `start` fails after `start` succeeded or after `status` says CRC is `Running`
7. ...

cfergeau committed:
go.mod: Update gvisor-tap-vsock to latest release

This brings go 1.20/1.21 support, but drops go 1.19 support.
After https://github.com/crc-org/crc/pull/3843 this should be fine.

This fixes this build error with go 1.21:
```
package github.com/crc-org/crc/v2/cmd/crc
        imports github.com/crc-org/crc/v2/cmd/crc/cmd
        imports github.com/containers/gvisor-tap-vsock/pkg/virtualnetwork
        imports github.com/containers/gvisor-tap-vsock/pkg/services/dhcp
        imports github.com/containers/gvisor-tap-vsock/pkg/tap
        imports gvisor.dev/gvisor/pkg/tcpip/stack
        imports gvisor.dev/gvisor/pkg/sync/locking
        imports gvisor.dev/gvisor/pkg/gohacks: build constraints exclude all Go files in /Users/yevhen/work/redhat/crc/vendor/gvisor.dev/gvisor/pkg/gohacks
make: *** [install] Error 1
```

praveenkumar commented:
/lgtm
/approve

praveenkumar commented:
/hold

It is failing for e2e with following error.
```
 go install -tags "containers_image_openpgp"  -ldflags="-X github.com/crc-org/crc/v2/pkg/crc/version.crcVersion=2.27.0 -X github.com/crc-org/crc/v2/pkg/crc/version.ocpVersion=4.13.12 -X github.com/crc-org/crc/v2/pkg/crc/version.okdVersion=4.13.0-0.okd-2023-09-03-082426 -X github.com/crc-org/crc/v2/pkg/crc/version.podmanVersion=4.4.4 -X github.com/crc-org/crc/v2/pkg/crc/version.microshiftVersion=4.13.12 -X github.com/crc-org/crc/v2/pkg/crc/version.commitSha=70fa0a "  ./cmd/crc
# gvisor.dev/gvisor/pkg/tcpip
vendor/gvisor.dev/gvisor/pkg/tcpip/tcpip.go:224:23: cannot convert a.addr[:4] (value of type []byte) to type [4]byte
vendor/gvisor.dev/gvisor/pkg/tcpip/tcpip.go:232:24: cannot convert a.addr[:16] (value of type []byte) to type [16]byte
note: module requires Go 1.20 
```

cfergeau commented:
> /hold
> 
> It is failing for e2e with following error.
> 
> ```
>  go install -tags "containers_image_openpgp"  -ldflags="-X github.com/crc-org/crc/v2/pkg/crc/version.crcVersion=2.27.0 -X github.com/crc-org/crc/v2/pkg/crc/version.ocpVersion=4.13.12 -X github.com/crc-org/crc/v2/pkg/crc/version.okdVersion=4.13.0-0.okd-2023-09-03-082426 -X github.com/crc-org/crc/v2/pkg/crc/version.podmanVersion=4.4.4 -X github.com/crc-org/crc/v2/pkg/crc/version.microshiftVersion=4.13.12 -X github.com/crc-org/crc/v2/pkg/crc/version.commitSha=70fa0a "  ./cmd/crc
> # gvisor.dev/gvisor/pkg/tcpip
> vendor/gvisor.dev/gvisor/pkg/tcpip/tcpip.go:224:23: cannot convert a.addr[:4] (value of type []byte) to type [4]byte
> vendor/gvisor.dev/gvisor/pkg/tcpip/tcpip.go:232:24: cannot convert a.addr[:16] (value of type []byte) to type [16]byte
> note: module requires Go 1.20 
> ```

With this change, we need go 1.20 or 1.21 everywhere

praveenkumar commented:
> With this change, we need go 1.20 or 1.21 everywhere

@cfergeau we already have go-1.20 everywhere as of now, even the openshift CI side?

cfergeau commented:
> > With this change, we need go 1.20 or 1.21 everywhere
> 
> @cfergeau we already have go-1.20 everywhere as of now, even the openshift CI side?

Seeing the e2e error, there must be one location which still has 1.19 or older.

cfergeau commented:
> > > With this change, we need go 1.20 or 1.21 everywhere
> > 
> > 
> > @cfergeau we already have go-1.20 everywhere as of now, even the openshift CI side?
> 
> Seeing the e2e error, there must be one location which still has 1.19 or older.

It's microshift e2e which is failing, and the logs show:
```
 Package golang-1.19.10-1.el9_2.x86_64 is already installed. 
```

cfergeau commented:
> > > > With this change, we need go 1.20 or 1.21 everywhere
> > > 
> > > 
> > > @cfergeau we already have go-1.20 everywhere as of now, even the openshift CI side?
> > 
> > 
> > Seeing the e2e error, there must be one location which still has 1.19 or older.
> 
> It's microshift e2e which is failing, and the logs show:
> 
> ```
>  Package golang-1.19.10-1.el9_2.x86_64 is already installed. 
> ```

Must be coming from https://github.com/openshift/release/blob/2b5d87adbb1d3922afdd051c7a489d03057a8376/ci-operator/step-registry/code-ready/crc/microshift/test/code-ready-crc-microshift-test-commands.sh#L83-L86

praveenkumar commented:
> > > > > With this change, we need go 1.20 or 1.21 everywhere
> > > > 
> > > > 
> > > > @cfergeau we already have go-1.20 everywhere as of now, even the openshift CI side?
> > > 
> > > 
> > > Seeing the e2e error, there must be one location which still has 1.19 or older.
> > 
> > 
> > It's microshift e2e which is failing, and the logs show:
> > ```
> >  Package golang-1.19.10-1.el9_2.x86_64 is already installed. 
> > ```
> 
> Must be coming from https://github.com/openshift/release/blob/2b5d87adbb1d3922afdd051c7a489d03057a8376/ci-operator/step-registry/code-ready/crc/microshift/test/code-ready-crc-microshift-test-commands.sh#L83-L86

you mean during https://github.com/openshift/release/blob/2b5d87adbb1d3922afdd051c7a489d03057a8376/ci-operator/step-registry/code-ready/crc/microshift/test/code-ready-crc-microshift-test-commands.sh#L46C3-L46C28 part because the `crc` binary already built using container and copied to VM but we do build e2e binary.

cfergeau commented:
> > > > > > With this change, we need go 1.20 or 1.21 everywhere
> > > > > 
> > > > > 
> > > > > @cfergeau we already have go-1.20 everywhere as of now, even the openshift CI side?
> > > > 
> > > > 
> > > > Seeing the e2e error, there must be one location which still has 1.19 or older.
> > > 
> > > 
> > > It's microshift e2e which is failing, and the logs show:
> > > ```
> > >  Package golang-1.19.10-1.el9_2.x86_64 is already installed. 
> > > ```
> > 
> > 
> > Must be coming from https://github.com/openshift/release/blob/2b5d87adbb1d3922afdd051c7a489d03057a8376/ci-operator/step-registry/code-ready/crc/microshift/test/code-ready-crc-microshift-test-commands.sh#L83-L86
> 
> you mean during https://github.com/openshift/release/blob/2b5d87adbb1d3922afdd051c7a489d03057a8376/ci-operator/step-registry/code-ready/crc/microshift/test/code-ready-crc-microshift-test-commands.sh#L46C3-L46C28 part because the `crc` binary already built using container and copied to VM but we do build e2e binary.

Yes, would be best to build everything on the same machine, and only copy pre-built binaries to gcloud.

anjannath commented:
(removed `has-to-be-in-release` as brew nodes are having `go 1.20` which is able to build without the issue this PR was fixing which happens with `go 1.21`)

cfergeau commented:
I created https://github.com/openshift/release/compare/master...cfergeau:release:prebuilt and updated this PR, but I don't know what is the best way forward for this openshift-release PR. Just file it?

praveenkumar commented on Makefile:
```diff
@@ -235,9 +235,6 @@ endif
-	VERSION_TO_TEST = --crc-version=$(CRC_VERSION)+$(COMMIT_SHA)
-endif
 e2e:
 	@go test --timeout=180m $(MODULEPATH)/test/e2e -tags "$(BUILDTAGS)" --ldflags="$(VERSION_VARIABLES)" -v $(PULL_SECRET_FILE) $(BUNDLE_LOCATION) $(CRC_BINARY) $(GODOG_OPTS) $(CLEANUP_HOME) $(VERSION_TO_TEST) $(INSTALLER_PATH) $(USER_PASSWORD)
```
should we remove ` $(VERSION_TO_TEST)` as part of e2e target?

cfergeau replied:
Since there is a `ifndef VERSION_TO_TEST` check, the code assumes `VERSION_TO_TEST` can be overridden externally (eg `VERSION_TO_TEST="--crc-version=0.1.2" make e2e`).
If we keep it, it will expand to the empty string if unset, which is fine, and the previous behaviour will be unchanged if it's set.
Imo it's better to keep it.

cfergeau commented:
I'm going to split this, first the CI changes, once they are merged, we can merge the corresponding openshift/release changes, and then this gvisor-tap-vsock PR.

cfergeau commented:
> I'm going to split this, first the CI changes, once they are merged, we can merge the corresponding openshift/release changes, and then this gvisor-tap-vsock PR.

Created https://github.com/crc-org/crc/pull/3876 with all the changes from this PR except for the gvisor-tap-vsock update.

cfergeau committed:
images: Update golang base image

There is no version of registry.access.redhat.com/ubi8/go-toolset with
golang 1.20:
```
$ podman build -f ./images/build-e2e/Dockerfile .
[1/2] STEP 1/5: FROM registry.access.redhat.com/ubi8/go-toolset:1.20 AS builder
Trying to pull registry.access.redhat.com/ubi8/go-toolset:1.20...
Error: creating build container: initializing source docker://registry.access.redhat.com/ubi8/go-toolset:1.20: reading manifest 1.20 in registry.access.redhat.com/ubi8/go-toolset: manifest unknown
```

This commit switches to
registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.20-openshift-4.15
which has golang 1.20 and is usable without authentication.
This stays on a rhel8 image as some of the test nodes are still using
rhel8, and building on a rhel9 node, and trying to run on a rhel8 node
is not possible (some libc linking issues at startup).

The builds will be noisy until
https://github.com/openshift-eng/art-tools/pull/115 is merged and gets
into these images.
If needed, we can set GO_COMPLIANCE_INFO=0 in the images environment to
make the build more quiet.

praveenkumar commented:
https://github.com/openshift/release/pull/44572 is created on openshift/release side since #3876 merged

praveenkumar commented:
/retest

cfergeau commented:
```
crc: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by crc)
crc: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by crc)
crc: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by crc)
```
CI fails because of this now :-/

cfergeau commented:
> ```
> crc: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by crc)
> crc: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by crc)
> crc: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by crc)
> ```
> 
> CI fails because of this now :-/

I'll try to use registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.20-openshift-4.15 instead of registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.15 (RHEL8 based instead of RHEL9 based) and see if this helps.

praveenkumar commented:
/retest

praveenkumar commented:
/retest
